src/java.base/share/classes/java/lang/ClassValue.java

```diff
@@ -369,12 +371,22 @@ private static ClassValueMap getMap(Class<?> type) {
     }
 
     private static final Object CRITICAL_SECTION = new Object();
+    private static final Unsafe UNSAFE = Unsafe.getUnsafe();
     private static ClassValueMap initializeMap(Class<?> type) {
         ClassValueMap map;
         synchronized (CRITICAL_SECTION) {  // private object to avoid deadlocks
             // happens about once per type
-            if ((map = type.classValueMap) == null)
-                type.classValueMap = map = new ClassValueMap();
+            if ((map = type.classValueMap) == null) {
+                map = new ClassValueMap();
+                // Place a Store fence as the last operation to emulate
```

Comment:
Since the store was moved i suggest the comment be slightly updated:
"Place a Store fence after construction and before publishing to emulate"

Reply:
Done.